### PR TITLE
Add hyper request smuggling vulnerability

### PR DESCRIPTION
**Pull request by DemiMarie-parity**

### CODE REVIEW

**Comment by Shnatsel** [comment]
Thanks for the report!

Please remove all the comments in the file (anything starting with `#`).

Also, I'd describe the effect as privilege escalation or ACL bypass. You need some other component to be vulnerable to get code execution.

File: crates/hyper/RUSTSEC-2020-0000.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Before you submit a PR using this template, **please delete the comments**
+# explaining each field, as well as any unused fields.
```
Comment (tarcieri) [inline crates/hyper/RUSTSEC-2020-0000.toml]: Please name this file `RUSTSEC-0000-0000.toml` (it's needed to pass the linter) and remove the comments.

**Comment by tarcieri** [comment]
Per @Shnatsel's comment on https://github.com/hyperium/hyper/issues/1925 I'll leave this PR open briefly to collect any additional comments or review

**Comment by tarcieri** [comment]
Assigned `RUSTSEC-2020-0008` in #256

File: crates/hyper/RUSTSEC-0000-0000.toml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+allowed, as demonstrated by sending a malformed HTTP request from a Substrate
+runtime.  This allows bypassing CORS restrictions.  In combination with other
+vulnerabilities, such as an exploitable web server listening on loopback, it may
+allow remote code execution.
+
+The flaw was corrected in hyper version 0.12.35.
+"""
+
+[versions]
+patched = [">= 0.12.35"]
```
Comment (hawkw) [inline crates/hyper/RUSTSEC-0000-0000.toml]: I believe the version listed here is incorrect; the vulnerability was fixed in [0.12.34](https://github.com/hyperium/hyper/blob/master/CHANGELOG.md#v01234-2019-09-04).

**Comment by jethrogb** [comment]
Are versions prior to 0.11 affected?

**Comment by Shnatsel** [comment]
Versions below 0.11 are not affected. The advisory was later updated with this info: https://github.com/RustSec/advisory-db/blob/master/crates/hyper/RUSTSEC-2020-0008.toml

**Comment by jethrogb** [comment]
Looks like https://rustsec.org/advisories/RUSTSEC-2020-0008.html hasn't been updated though.

**Comment by tarcieri** [comment]
Should be updated now. Sorry about that.

(I should really look into automating the web site generation with GitHub Actions)

**Pull request merged**